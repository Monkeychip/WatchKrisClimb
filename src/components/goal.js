import React from 'react';
import { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions/actions_index'; 
import { Button, Form } from 'semantic-ui-react'

//potential to change this to a const, but doing in component first in case need to change state
class Goal extends Component {
 render(){
  return(
    <Form size='large'key ='large'>
    <Form.Field>
      <label>Enter Elevation Goal for the year</label>
      <input type='number' max={2000000} placeholder="50,000 ft" width={2}  />
    </Form.Field>
    <Button type='submit'>Submit</Button>
    </Form>
  )
 }
}

export default Goal