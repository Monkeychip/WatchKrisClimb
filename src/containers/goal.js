import React from 'react';
import { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Form } from 'semantic-ui-react';
import { Field, reduxForm} from 'redux-form'; 
import * as actions from '../actions/actions_index';


let Goal = props => {

  const { handleSubmit } = props

  const formatter = {
    number: value => value ? (+value).toLocaleString() : ''
  };

  const parser = {
    number: value => value ? (value.match(/\d+/g) || []).join('') : ''
  };

  return (
    <form onSubmit={handleSubmit} size='large' key='large'>
		<div className="field">
          <div id="elevation-label"><label>Enter Elevation Goal for the Year:</label></div>

                <div className="ui right labeled input" >
                    <Field
                        name="number"                   
                        type="number"
                        max={2000000} 
                        placeholder="50,000 ft"
                        component="input"
                       	className="two wide field"
                    />
                    <div className="ui basic label" id="fix-ft-label">
				      		ft
				    </div>
                </div>
          </div>
          <button type="submit" id="submit-button" className="ui inverted blue button ">
            Show Goal
          </button>
    </form>
  )
}

Goal = reduxForm({
  form: 'goal'
})(Goal)

export default Goal

