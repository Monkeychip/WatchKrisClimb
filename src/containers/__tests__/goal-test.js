import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Goal } from "../goal";

Enzyme.configure({adapter: new Adapter()});

test("goal renders correctly", () => {
  const component = shallow(<Goal />)
  expect(component).toMatchSnapshot();
});

