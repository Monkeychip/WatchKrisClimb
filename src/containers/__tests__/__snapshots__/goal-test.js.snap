// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Component snapshot should render the snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
  size="large"
>
  <div
    className="field"
  >
    <div
      id="elevation-label"
    >
      <label>
        Enter Elevation Goal for the Year:
      </label>
    </div>
    <div
      className="ui right labeled input"
    >
      <div
        className="ui right labeled input"
      >
        <input
          className="two wide field"
          max={2000000}
          name="number"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="50,000 ft"
          type="number"
          value=""
        />
      </div>
      <div
        className="ui basic label"
        id="fix-ft-label"
      >
        ft
      </div>
    </div>
  </div>
  <button
    className="ui inverted blue button "
    id="submit-button"
    type="submit"
  >
    Show Goal
  </button>
</form>
`;

exports[`Testing that the redux-form matches snapshot should render the snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
  size="large"
>
  <div
    className="field"
  >
    <div
      id="elevation-label"
    >
      <label>
        Enter Elevation Goal for the Year:
      </label>
    </div>
    <div
      className="ui right labeled input"
    >
      <div
        className="ui right labeled input"
      >
        <input
          className="two wide field"
          max={2000000}
          name="number"
          onBlur={[Function]}
          onChange={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          placeholder="50,000 ft"
          type="number"
          value=""
        />
      </div>
      <div
        className="ui basic label"
        id="fix-ft-label"
      >
        ft
      </div>
    </div>
  </div>
  <button
    className="ui inverted blue button "
    id="submit-button"
    type="submit"
  >
    Show Goal
  </button>
</form>
`;

exports[`goal component renders correctly 1`] = `
<Connect(Form(Goal))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="goal"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;

exports[`goal renders correctly 1`] = `
<Connect(Form(Goal))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="goal"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
